src/components/HomePage.jsx

```diff
@@ -1,19 +1,225 @@
+import PlaceholderLoader from '../views/placeholderLoader.jsx';
+import { getAverageRatings, totalRatings } from '../utils/getAverageRatings';
 
-class HomePage extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
+/**
+ * @description - returns the home page view
+ * @param {props} articles - all authors article
+ * @param {props} articlesCategory - articles categories
+ * @return {JSX} 
+ */
+export class HomePage extends Component {
+  state = {
+    fiveStarAuthors: [],
   }
+  async componentDidMount() {
+    const { getArticles, getArticlesCategory } = this.props;
+    await Promise.all([getArticles(), getArticlesCategory()]);
+  }
+
+  /**
+   * @description Gets the averahe ratings of author's article
+   * @param {object} article - each authors article
+   * @return {array} fiveStaAauhtors - author's rated highly
+   */
+  averageRatings = (article) => {
+    getAverageRatings(article);
+    totalRatings.map(async rating => {
+      if (rating.averageRatings === 5) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      } else if (rating.averageRatings === 4) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      } else if (rating.averageRatings === 3) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      } else if (rating.averageRatings === 2) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      } else if (rating.averageRatings === 1) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      }
+    });
+  }
+
+  /**
+ * @description set the state with start authors
+ * @return {void}
+ */
+  setAuthorsState = (rating) => (this.setState({
+    fiveStarAuthors: [...this.state.fiveStarAuthors, rating],
+  }));
+
+  /**
+ * @description found uthors
+ * @param {object} rating - each authors article ratings
+ * @return {array}
+ */
+  found = (rating) => (this.state.fiveStarAuthors.find(fiveStarAuthors => {
+    return rating.authorsId === fiveStarAuthors.authorsId;
+  }));
+
+  /**
+ * @description Gets the recommended authors
+ * @param {object} article - each authors article
+ * @return {array} fiveStaAauhtors - author's rated highly
+ */
+  recommendedAuthor(article) {
+    this.averageRatings(article);
+  }
+
   render() {
+    const { articles, articlesCategory, loadingArticles, loadingCategory } = this.props;
+    
+    articles.map(article => {
+      this.recommendedAuthor(article);
+    });
+
     return (
       <div>
-        <p>Hello World</p>
-        <a Link="/login" href="/login">Login </a>
+        <Navbar
+          categories={'Categories'}
+          authors={'Authors'}
+          bookmark={'Bookmarks'}
+          search={'Search'}
+          write={'Write an areticle'}
+          login={'Login'}
+        />
+
+        <main className="main-body">
```

Comment:
#### Nitpick
- Resolve this if you have already created the view file for this.

Reply:
Alright